src/main/java/beam/physsim/cch/RoutingFrameworkTravelTimeCalculator.scala

```diff
@@ -0,0 +1,230 @@
+package beam.physsim.cch
+
+import java.util
+import java.util.concurrent.Executors
+
+import beam.agentsim.events.PathTraversalEvent
+import beam.sim.BeamServices
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.geom.Coordinate
+import org.apache.commons.lang.time.StopWatch
+import org.matsim.api.core.v01.network.Link
+import org.matsim.core.controler.events.IterationEndsEvent
+
+import scala.collection.JavaConverters._
+import scala.collection.convert.ImplicitConversionsToScala._
+import scala.collection.mutable
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+class RoutingFrameworkTravelTimeCalculator(
+  private val beamServices: BeamServices,
+  private val osmInfoHolder: OsmInfoHolder,
+  private val routingFrameworkWrapper: RoutingFrameworkWrapper,
+) extends LazyLogging {
+
+  private implicit val executionContext: ExecutionContext = ExecutionContext.fromExecutorService(
+    Executors.newFixedThreadPool(
+      Math.max(Runtime.getRuntime.availableProcessors() / 4, 1),
+      new ThreadFactoryBuilder().setDaemon(true).setNameFormat("routing-framework-worker-%d").build()
+    )
+  )
+
+  private val odsFactor: Int =
+    Math.max(
+      (1.0 / beamServices.beamConfig.beam.agentsim.agentSampleSizeAsFractionOfPopulation *
+      beamServices.beamConfig.beam.physsim.cch.congestionFactor).toInt,
+      1
+    )
+
+  private val graph: RoutingFrameworkGraph = routingFrameworkWrapper.generateGraph()
+
+  private val coord2VertexId: Map[Coordinate, Long] =
+    graph.vertices.map(x => x.coordinate -> x.id).toMap
+
+  private val coordKey = (coord: Coordinate) => (coord.x * 10).toInt + "|" + (coord.y * 10).toInt
+
+  private val coordKey2Coordinates: Map[String, Seq[Vertex]] = graph.vertices
+    .map(v => coordKey(v.coordinate) -> v)
+    .groupBy(_._1)
+    .mapValues(_.map(_._2))
+
+  def getLink2TravelTimes(
+    pathTraversalEvents: util.Collection[PathTraversalEvent],
+    iterationEndsEvent: IterationEndsEvent,
+    links: util.Collection[_ <: Link],
+    maxHour: Int
+  ): java.util.Map[String, Array[Double]] = {
+    val iterationNumber: Int = iterationEndsEvent.getIteration
+    val startTime: Long = System.currentTimeMillis
+
+    logger.info("Finished creation of graph {}", System.currentTimeMillis - startTime)
```

Comment:
This doesn't do anything as far as I can tell. The total time is not going to be useful.

Reply:
Also, why go with stopwatch elsewhere but not here?

Reply:
removed